config/routes.rb

```diff
@@ -36,4 +36,11 @@
   get '/shelters/:shelter_id/reviews/new', to: 'reviews#new'
   post '/shelters/:shelter_id', to: 'reviews#create'
   delete '/shelters/:shelter_id/reviews/:id', to: 'reviews#destroy'
+
+  # favorites
+  get '/favorites', to: 'favorites#index'
+
+  patch '/favorites/:pet_id', to: 'favorites#update'
+  delete '/favorites', to: 'favorites#destroy_all'
+  delete '/favorites/:pet_id', to: 'favorites#destroy'
```

Comment:
What if we combine these and add a conditional to the controller's #destroy action? If params[:pet_id] is nil, destroy all, otherwise destroy one.

Reply:
sure I will push the new code up